pymc_experimental/model_builder.py

```diff
@@ -601,17 +601,16 @@ def sample_prior_predictive(
 
         if self.model is None:
             self.build_model(X_pred, y_pred)
-
-        self._data_setter(X_pred, y_pred)
-        if self.model is not None:
-            with self.model:  # sample with new input data
-                prior_pred: az.InferenceData = pm.sample_prior_predictive(samples, **kwargs)
-                self.set_idata_attrs(prior_pred)
-                if extend_idata:
-                    if self.idata is not None:
-                        self.idata.extend(prior_pred)
-                    else:
-                        self.idata = prior_pred
+        else:
+            self._data_setter(X_pred, y_pred)
+        with self.model:  # sample with new input data
+            prior_pred: az.InferenceData = pm.sample_prior_predictive(samples, **kwargs)
+            self.set_idata_attrs(prior_pred)
+            if extend_idata:
+                if self.idata is not None:
+                    self.idata.extend(prior_pred)
```

Comment:
I think we want this so that it overrides pre-existing prior_predictive groups. Can you confirm in a test?

```suggestion
                    self.idata.extend(prior_pred, join="right")
```

Reply:
There are might be some other calls to `extend` in other methods that should have `join="right"` added. I'll check, and if there are, I'll add a commit to add that.